cli/command/volume/prune.go

```diff
@@ -41,18 +45,37 @@ func NewPruneCommand(dockerCli command.Cli) *cobra.Command {
 	}
 
 	flags := cmd.Flags()
+	flags.BoolVarP(&options.all, "all", "a", false, "Remove all unused volumes, not just anonymous ones")
+	flags.SetAnnotation("all", "version", []string{"1.42"})
 	flags.BoolVarP(&options.force, "force", "f", false, "Do not prompt for confirmation")
 	flags.Var(&options.filter, "filter", `Provide filter values (e.g. "label=<label>")`)
 
 	return cmd
 }
 
-const warning = `WARNING! This will remove all local volumes not used by at least one container.
+const (
+	unusedVolumesWarning = `WARNING! This will remove all local anonymous volumes not used by at least one container.
```

Comment:
Can we change this message around a little so that the all vs non-all difference is more obvious?

```suggestion
	unusedVolumesWarning = `WARNING! This will remove anonymous local volumes not used by at least one container.
```

For comparison, the "all" message:

```go
	allVolumesWarning = `WARNING! This will remove all local volumes not used by at least one container.
```

Reply:
Ah, yes, that's better.

I was even considering if we need the  warning at all for the anonymous volumes (and only a warning for the `--all` case), but left it for now